electrumsv/tests/test_wallet.py

```diff
@@ -486,128 +508,346 @@ def test_multisig(self, mock_app_state, tmp_storage) -> None:
+            assert len(wallet.get_accounts()) == 2
+            private_key_account = cast(ImportedPrivkeyAccount,
+                [ entry for entry in wallet.get_accounts() if not entry.is_petty_cash() ][0])
+            private_key_keystore = cast(Imported_KeyStore, private_key_account.get_keystore())
+            # Pre-decrypt the prv for later comparison so the initial password is not needed there.
+            for public_key, encrypted_prv in private_key_keystore._keypairs.items():
+                keyinstance_private_keys_after[public_key.to_hex()] = pw_decode(encrypted_prv,
+                    new_password)
+        assert keyinstance_private_keys_after == keyinstance_private_keys
+
+        # Derivation data in the MasterKeys table.
+        master_key_xprvs_after: list[str] = []
+        master_key_seeds_after: list[str] = []
+        master_key_passphrases_after: list[str] = []
+        for keystore in wallet.get_keystores():
+            data = keystore.to_derivation_data()
+            for entry_name in ("seed", "passphrase", "xprv"):
+                entry_value = cast(str | None, data.get(entry_name, None))
+                if not entry_value:
+                    continue
+                decoded_value = pw_decode(entry_value, new_password)
+                if entry_name == "seed":
+                    master_key_seeds_after.append(decoded_value)
+                    self.re_encryption_test_state["count_coverage_masterkeys_table_seed"] += 1
+                if entry_name == "xprv":
+                    master_key_xprvs_after.append(decoded_value)
+                    self.re_encryption_test_state["count_coverage_masterkeys_table_xprv"] += 1
+                if entry_name == "passphrase":
+                    master_key_passphrases_after.append(decoded_value)
+                    self.re_encryption_test_state["count_coverage_masterkeys_table_passphrase"] += 1
+        assert master_key_xprvs_after == master_key_xprvs
+        assert master_key_seeds_after == master_key_seeds
+        assert master_key_passphrases_after == master_key_passphrases
+
+        # Newer wallets have a `Servers` table with an `encrypted_api_key` column
+        encrypted_api_keys_after = list[str]()
+        for server in wallet.data.read_network_servers():
+            if server.encrypted_api_key is not None:
+                self.re_encryption_test_state['count_coverage_server_api_key'] += 1
+                decoded_value = pw_decode(server.encrypted_api_key, new_password)
+                encrypted_api_keys_after.append(decoded_value)
+        assert encrypted_api_keys_after == encrypted_api_keys
+
+        # passed the test assertions
+        self.re_encryption_test_state['failing_wallets'].remove(storage_info.filename)
+        self.re_encryption_test_state['total_wallets_tested'] += 1
+        # This is the last parameter in the set
+        if self.re_encryption_test_state['total_wallets_tested'] == total_expected_wallets:
```

Comment:
No. This should not be part of the test. It should happen after the parametrized tests are complete as discussed. And also all this printing is unnecessary and adds complexity. We should just assert in the post step that the expected cases have happened.

Reply:
It might be as simple as moving this to a second test defined after this test that asserts if any of the dataclass fields are 0.